certbot/certbot/_internal/client.py

```diff
@@ -454,7 +454,22 @@ def obtain_certificate(self, domains: List[str], old_keypath: Optional[str] = No
                 os.remove(csr.file)
             return self.obtain_certificate(successful_domains)
         else:
-            cert, chain = self.obtain_certificate_from_csr(csr, orderr)
+            try:
+                cert, chain = self.obtain_certificate_from_csr(csr, orderr)
+            except messages.Error as error:
+                if self.config.allow_subset_of_names and error.subproblems is not None:
+                    failed_domains = [problem.identifier.value for problem in error.subproblems
+                                        if problem.identifier is not None]
+                    successful_domains = [x for x in domains if x not in failed_domains]
+                    if successful_domains != domains:
+                        logger.info("error obtaining certificate with --allow-subset-of-names "
```

Comment:
In my opinion, the `logger.info` call should be promoted to `display_util.notify`. 

When testing this, I felt uneasy that, by default, the user is not informed that the certificate coverage has shrunk. 

I think the error could also benefit from some prettifying. Perhaps:

>Unable to obtain certificate with every requested domain. Retrying without: %s.

Reply:
Changed to use display_util.notify 🙂